test/jdk/javax/swing/JSpinner/4670051/JSpinnerFieldUnderCursorTest.java

```diff
@@ -0,0 +1,302 @@
+ * exactly spins the field where cursor is there.
+ * @run main JSpinnerFieldUnderCursorTest
+ */
+public class JSpinnerFieldUnderCursorTest {
+
+    private static final Calendar cal1 = Calendar.getInstance();
+    private static final Calendar cal2 = Calendar.getInstance();
+    private static Robot robot;
+    private static JSpinner spinner;
+    private static Date initValue;
+    private static Date upValue;
+    private static Date downValue;
+    private static JFrame frame;
+    private static boolean passed = true;
+    private static volatile Point spinnerUpButtonCenter;
+    private static volatile Point spinnerDownButtonCenter;
+    private static volatile Date spinnerValue;
+
+    private static void createUI() {
+        frame = new JFrame();
+        JPanel panel = new JPanel();
+
+        spinner = new JSpinner(new SpinnerDateModel());
+        JSpinner.DateEditor editor = new JSpinner.DateEditor(spinner, " dd/MM/yy ");
+        spinner.setEditor(editor);
+        spinner.setComponentOrientation(ComponentOrientation.LEFT_TO_RIGHT);
+        panel.add(spinner);
+
+        frame.add(panel);
+        frame.setUndecorated(true);
+        frame.pack();
+        frame.setAlwaysOnTop(true);
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+
+    }
+
+    private static void setLookAndFeel(final String laf) {
+        try {
+            UIManager.setLookAndFeel(laf);
+            System.out.println("LookAndFeel: " + laf);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void runTest() throws Exception {
+        robot = new Robot();
+        robot.setAutoWaitForIdle(true);
+        robot.setAutoDelay(100);
+        List<String> lafs = Arrays.stream(getInstalledLookAndFeels())
+                                  .map(UIManager.LookAndFeelInfo::getClassName)
+                                  .collect(Collectors.toList());
+        for (final String laf : lafs) {
+            try {
+                SwingUtilities.invokeAndWait(() -> {
+                    setLookAndFeel(laf);
+                    createUI();
+                });
+                SwingUtilities.invokeAndWait(() -> {
+                    Point loc = spinner.getLocationOnScreen();
+                    int editorWidth = spinner.getEditor().getWidth();
+                    int buttonWidth = spinner.getWidth() - editorWidth;
+                    int quarterHeight = spinner.getHeight() / 4;
+
+                    spinnerUpButtonCenter = new Point(loc.x + editorWidth
+                            + (buttonWidth / 2),
+                            loc.y + quarterHeight);
+                    spinnerDownButtonCenter = new Point(spinnerUpButtonCenter.x,
+                            loc.y + (3 * quarterHeight));
+                });
+
+                // Cursor at Day field.
+                updateSpinnerValue();
+                // Increment Day
+                initValue = spinnerValue;
```

Comment:
May I suggest modifying `updateSpinnerValue` so that it returns the updated value? I think this code will be clearer:
```java
                initValue = getSpinnerValue();
                // Increment Day
```